### PR TITLE
cardano-api: 8.38 -> 9.1

### PR DESCRIPTION
**Pull request by locallycompact**
Working with the CI team I suggested to drop 8.10 and 9.2 in this branch. Would there be any objections to that? We can support 9.10 shortly.

### CODE REVIEW

File: quickcheck-threatmodel/src/Test/QuickCheck/ThreatModel/Cardano/Api.hs [hunk 1]
```diff
@@ -80,16 +81,16 @@ recomputeScriptData _ _ TxBodyNoScriptData = TxBodyNoScriptData
 recomputeScriptData i f (TxBodyScriptData era dats (Ledger.Redeemers rdmrs)) =
   TxBodyScriptData era dats
     (Ledger.Redeemers $ Map.mapKeys updatePtr $ Map.filterWithKey idxFilter rdmrs)
-  where -- updatePtr = Ledger.hoistPlutusPurpose (\(Ledger.AsIndex ix) -> Ledger.AsIndex (f ix)) -- TODO: replace when hoistPlutusPurpose is available
+  where -- updatePtr = Ledger.hoistPlutusPurpose (\(Ledger.AsIx ix) -> Ledger.AsIx (f ix)) -- TODO: replace when hoistPlutusPurpose is available
```
Comment (noonio) [inline quickcheck-threatmodel/src/Test/QuickCheck/ThreatModel/Cardano/Api.hs:84]: Can you work out how to use this function `hoistPlutusPurpose` ? I note it exists now ( I can't quite work it out 😅 )

File: quickcheck-threatmodel/src/Test/QuickCheck/ThreatModel/Cardano/Api.hs [hunk 2]
```diff
@@ -229,6 +231,9 @@ validateTx pparams tx utxos = case result of
     systemStart :: SystemStart
     systemStart = SystemStart $ posixSecondsToUTCTime 0
 
+    genesisWindow :: GenesisWindow
+    genesisWindow = GenesisWindow 10
```
Comment (noonio) [inline quickcheck-threatmodel/src/Test/QuickCheck/ThreatModel/Cardano/Api.hs:235]: Why is `10` a reasonable choice here?

**Review by noonio — APPROVED** [review]
Looks reasonable to me!

**Review by RSoulatIOHK — APPROVED** [review]
Looks good. I just have the same question as @noonio about the genesis window value

File: quickcheck-threatmodel/src/Test/QuickCheck/ThreatModel/Cardano/Api.hs (hunk 2 above)
Comment (locallycompact) [reply quickcheck-threatmodel/src/Test/QuickCheck/ThreatModel/Cardano/Api.hs:235]: Not sure. There were several default fixtures in various places. Here is one of them.

https://github.com/IntersectMBO/ouroboros-consensus/blob/main/ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HardFork/History/EraParams.hs#L159

**Pull request merged**